spec/helpers/search_helper_spec.rb

```diff
@@ -1,6 +1,68 @@
 require 'rails_helper'
 
 RSpec.describe SearchHelper, type: :helper do
+
+  describe "#paginate" do
+
+    let (:page_params) {{
+      :total_pages=>3,
+      :previous_link=>"/petitions?page=1&state=all",
+      :next_link=>"/petitions?state=all",
+      "model" => Petition
+    }}
+
+    let (:first_page_params) {{
+      "first_page?" => true,
+      "last_page?" => false,
+      "previous_params" => {:state=>:all, :page=>nil},
+      "next_params" => {:state=>:all, :page=>2},
+      :previous_page=>nil,
+      :next_page=>2,
+    }}
+
+    let (:intermediary_page_params) {{
```

Comment:
I think calling this params is misleading - they're stubbed methods on the petitions double. I'd prefer to see them inside the context block for each case so that they're grouped together lexically and not composed which masks what the actual stubs are.